test/scripts/e2e_subs/min_balance.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python
+
+from base64 import b64decode
+from datetime import datetime
+from pathlib import PurePath
+import sys
+
+import algosdk.future.transaction as txn
+
+from goal import Goal
+
+# Set INTERACTIVE True if you want to run a remote debugger interactively on the given PORT
+INTERACTIVE, DEBUGPORT = False, 4312
+
+
+def initialize_debugger():
+    import multiprocessing
+
+    if multiprocessing.current_process().pid > 1:
+        import debugpy
+
+        debugpy.listen(("0.0.0.0", DEBUGPORT))
+        print("Debugger is ready to be attached, press F5", flush=True)
+        debugpy.wait_for_client()
+        print("Visual Studio Code debugger is now attached", flush=True)
+
+
+if INTERACTIVE:
+    initialize_debugger()
+
+
+CONSENSUS_MIN_BALANCE = 100_000
+ASSET_MIN_BALANCE = 100_000
+APP_MIN_BALANCE = 100_000
+OPTIN_MIN_BALANCE = 100_000
+# app schemas:
+APP_KV_MIN_BALANCE = 25_000
+APP_INTS_MIN_BALANCE = 3_500
+APP_BYTES_MIN_BALANCE = 25_000
+
+# per userBalance.go::MinBalance() as of Dec 2021:
+EXTRA_PAGE_MIN_BALANCE = APP_MIN_BALANCE
+
+TEAL = f"""#pragma version 6
```

Comment:
@jannotti - simpler TEAL program not using ABI

Reply:
I'd use version 5 here.  We're probably going to release this before teal 6 is available.  Nothing will break, since this is only a test, but it seems weird to test it with a v6 teal script even though it's a feature that comes out before v6.